src/helpers/ring.rs

```diff
@@ -127,18 +130,18 @@ pub mod mock {
             }
         }
 
-        fn set_left(&mut self, left: Sender<MessageEnvelope>) {
+        fn set_left(&mut self, left: Sender<MessageEnvelope<S>>) {
             self.left = Some(left);
         }
 
-        fn set_right(&mut self, right: Sender<MessageEnvelope>) {
+        fn set_right(&mut self, right: Sender<MessageEnvelope<S>>) {
             self.right = Some(right);
         }
     }
 
     #[async_trait]
-    impl Ring for TestHelper {
-        async fn send<T: Message>(&self, dest: HelperAddr, msg: T) -> Result<(), Error> {
+    impl<S: Step> Ring<S> for TestHelper<S> {
+        async fn send<T: Message>(&self, dest: HelperAddr, step: S, msg: T) -> Result<(), Error> {
```

Comment:
IMPORTANT: I don't see anything here that ensures that sending occurs in order.

Reply:
I thought the idea was to let actual implementation of this trait to reorder events as they arrive according to their position (query + record + step). Is there a way to impose ordering on caller side without creating a custom scheduler?

Reply:
This is the implementation, so this is going to need to deal with that somehow.

What we discussed was a unique stream of items for each (query + step), but that each item would have a counter (or index) that would place it in sequence on that stream.

Reply:
gotcha. I thought you had questions about `send` interface. 

I was hoping to push it separately to unblock folks faster, but I can do it as part of this change too

Reply:
Your call.  Though I think that your tests were hiding any potential problems that might come about from not having implemented it, so I wanted to ensure that people wouldn't use it correctly and then be surprised when they got garbage sometimes (it should be rare that things get reordered if you aren't working very hard, but it could happen).